Dockerfile

```diff
@@ -1,28 +1,38 @@
-FROM node:10-alpine
+FROM node:10-alpine as base
 LABEL Description="Web Frontend of the Social Network Human-Connection.org" Vendor="Human-Connection gGmbH" Version="0.0.1" Maintainer="Human-Connection gGmbH (developer@human-connection.org)"
 
 # Expose the app port
-EXPOSE 3000
-
+ARG BUILD_COMMIT
+ENV BUILD_COMMIT=$BUILD_COMMIT
 ARG WORKDIR=/nitro-web
 RUN mkdir -p $WORKDIR
 WORKDIR $WORKDIR
 
 # See: https://github.com/nodejs/docker-node/pull/367#issuecomment-430807898
 RUN apk --no-cache add git
 
-# Install Web Application
-COPY package.json .
-COPY yarn.lock .
-COPY styleguide/ ./styleguide
-RUN yarn install --production=false --frozen-lockfile --non-interactive
+COPY locales ./locales
```

Comment:
The copy part belongs into the builder where you copy stuff then over.

Reply:
Similar to my other comment above, we cannot copy anything in the `build-and-test` stage from host, it must first be copied from host to `base`, which we then use in both the `build-and-test` stage and `production` stage

Reply:
Thats not true to my knowladge, you can copy files and folder from one image to another without the need to extend it. `COPY <src> <dest> --from=builder`

Reply:
For some reference: https://glebbahmutov.com/blog/making-small-docker-image/

Reply:
That is possibly a very nice solution; however, that is different from the Nitro-Backend, which you indicated should be used as a reference. Also, the link you shared does use the base image for both TEST and PROD... the only difference I guess is that they copy everything from host to the base image, which we can certainly try, but I would be surprised if reduced the size of the image drastically.

It's not really clear to me how these suggestions help to solve the failing build, but I can certainly give it a go, and if it create a larger image that reliably passes the build, it would be a good first step.

Reply:
I just try to give some input, might not be the best as I’m doing it from the go so I’m not involved 100%, sorry for that if it’s not helpful, just ignore it if it does not make sense.

Reply:
I think it very much is helpful! I am a bit stuck on why it's failing in the way it's failing, and having ideas of things to try is always welcome! thanks for taking the time to try to debug it... I will push it up and see what happens. Maybe we get lucky